File: web/src/types/customer.ts
/* ========================================
   顾客模块类型（对齐 OpenAPI 契约）
   ======================================== */

/** 送货地址（响应） */
export interface ShipAddress {
  id?: number;
  address: string;
  remark?: string | null;
}

/** 送货地址输入（创建/更新） */
export interface ShipAddressInput {
  address: string;
  remark?: string;
}

/** 顾客详情（对齐 Customer schema） */
export interface Customer {
  id: number;
  code: string;
  name: string;
  address: string;
  /** 收/发货地址列表（连锁客户可有多个） */
  shipAddresses: ShipAddress[];
  contact?: string | null;
  phone?: string | null;
  email?: string | null;
  remark?: string | null;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/** 顾客列表项（对齐 CustomerSummary schema） */
export interface CustomerSummary {
  id: number;
  code: string;
  name: string;
  address: string;
  shipAddresses: ShipAddress[];
  contact?: string | null;
  phone?: string | null;
  status: 0 | 1;
  createdAt: string;
}

/** 顾客列表分页响应 */
export interface CustomerListResponse {
  items: CustomerSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

/** 创建顾客请求 */
export interface CreateCustomerRequest {
  code: string;
  name: string;
  address: string;
  shipAddresses: ShipAddressInput[];
  contact?: string;
  phone?: string;
  email?: string;
  remark?: string;
  status?: 0 | 1;
}

/** 更新顾客请求（code 不可改，状态走独立接口；shipAddresses 提供则整列表替换） */
export interface UpdateCustomerRequest {
  name?: string;
  address?: string;
  shipAddresses?: ShipAddressInput[];
  contact?: string;
  phone?: string;
  email?: string;
  remark?: string;
}

/** 状态变更请求 */
export interface UpdateCustomerStatusRequest {
  status: 0 | 1;
}

/** 顾客查询参数 */
export interface CustomerQueryParams {
  keyword?: string;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}

File: api/src/main/java/com/aistore/module/customer/entity/Customer.java
package com.aistore.module.customer.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 顾客（客户公司）实体
 * 对应数据库表 customer
 * 收/发货地址为一对多，存于 customer_ship_address 子表（见 {@link CustomerShipAddress}）
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("customer")
public class Customer {

    /**
     * 主键 ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 客户编码（创建后不可修改，全局唯一）
     */
    private String code;

    /**
     * 客户公司名称（全局唯一）
     */
    private String name;

    /**
     * 客户公司地址
     */
    private String address;

    /**
     * 联系人
     */
    private String contact;

    /**
     * 联系电话
     */
    private String phone;

    /**
     * 邮箱
     */
    private String email;

    /**
     * 备注
     */
    private String remark;

    /**
     * 状态：0=禁用，1=启用
     */
    private Integer status;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 创建人 ID
     */
    private Long createdBy;

    /**
     * 更新人 ID
     */
    private Long updatedBy;

    /**
     * 逻辑删除：0=未删除，1=已删除
     */
    @TableLogic
    private Integer deleted;
}
